mystery-deckbuilder/Assets/Scripts/Encounter/Filter.cs

```diff
@@ -0,0 +1,49 @@
+
+
+using System;
+using UnityEngine;
+
+public static class Filters
+{
+    public static Array GetFilterByID(int id, int baseComp, int basePatience, string element) //param list: card element, base compliance, base patience. can use GameState for a lot of access stuff I guess 
+    {
+        switch (id)
+        {
+            case 0:
+                return NoEffect();
+            case 1:
+                return WeaknessCheck(baseComp, element);
+            case 2:
+                return ResistanceCheck(baseComp, element);
+            default:
+                return null;
+        }
+    }
```

Comment:
NIT: This strategy is okay, although I personally believe that we should just be passing a reference to a card. All of these features (and future features) could be accessed quite easily